src/ba-transport.c

```diff
@@ -845,11 +849,11 @@ struct ba_transport *ba_transport_new_a2dp(
 	t->acquire = transport_acquire_bt_a2dp;
 	t->release = transport_release_bt_a2dp;
 
-	ba_transport_set_codec(t, codec->codec_id);
-
 	storage_pcm_data_sync(&t->a2dp.pcm);
 	storage_pcm_data_sync(&t->a2dp.pcm_bc);
 
+	ba_transport_set_codec(t, codec->codec_id);
```

Comment:
Why this reorder is required?

Reply:
I think its a hangover from previous draft - now reverted